javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,112 @@
+let personName = null;
+let toDo = [];
+
+function getReply(command) {
+
+    //----------My name command-------------
+    if (command.includes("Hello my name is")) {
+        personName = command.split(' ').splice(-1);
+        return "Nice to meet you  " + personName;
+    }
+
+    //------------- What is my name command--------
+    if (command.includes("What is my name")) {
+        if (personName !== null) {
+            return "Your name is " + personName;
+
+        } else {
+            return "You did not say your name";
+        }
+    }
+    //------------- Adding todos -------------
+    if (command.includes("Add")) {
+        let todoItem = command.split('Add').pop().split('to my todo').splice(0, 1);
+        todoItem = todoItem.toString().trim();
+        toDo.push(todoItem);
+        return todoItem + " added to your todo";
+    }
+
+    //-------------Remove todo command-----------
+    if (command.includes("Remove")) {
+        let todoItem = command.split('Remove').pop().split('from my todo').splice(0, 1);
+        todoItem = todoItem.toString().trim();
+        const index = toDo.indexOf(todoItem);
+        if (index > -1) {
+            toDo.splice(index, 1);
+            return "Removed " + todoItem + " from your TodoList";
+
+        } else {
+            return todoItem + " not found in your todo-list";
+        }
+    }
+
+    //----- return all todos-----------
+    if (command.includes("on my todo?")) {
+
+        if (toDo.length > 1) {
+            let noOfItems = toDo.length;
+            let lastItem = toDo.pop(); //Last item removed from the toDo array 
+            return "You have " + noOfItems + " items-" + toDo + " and " + lastItem;
+        }
+        if (toDo.length == 1) {
```

Comment:
Try always to avoid using `==`. Even the 2 sides of it are of different data types use parser then compare with `===`.